pkg/apis/servicecatalog/types.go

```diff
@@ -46,6 +46,30 @@ type ClusterServiceBrokerList struct {
 	Items []ClusterServiceBroker
 }
 
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ServiceBroker represents an entity that provides ServiceClasses for use in the
+// service catalog. ServiceBroker is backed by an OSBAPI v2 broker supporting the
+// latest minor version of the v2 major version.
+type ServiceBroker struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	Spec   ServiceBrokerSpec
+	Status ServiceBrokerStatus
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```

Comment:
I think the deepcopy-gen hook only needs to be added once for the top of the file, is that true? If so then I think you can remove these as it is already on line 26. I don't know if it only applies to the following struct or all structs in the file so please educate me.

Reply:
I might have copied the comment block from the top, can someone confirm? I'd like to know as well. Also, are these hooks documented somewhere?

Reply:
I believe that you want deepcopy gen for objects for anything that could be watched for individually, i.e. a root level object. So yes for ServiceBroker, but not necessary for say ServiceBrokerStatus. It helps people make a copy before mutating an object pulled from the cache.